### PR TITLE
BUG: ValueError: Cannot convert non-finite values (NA or inf) to integer only when DF exceed certain size

### LINKED ISSUE #35227: BUG: ValueError: Cannot convert non-finite values (NA or inf) to integer only when DF exceed certain size
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### 

Here is the code in question:

```
    print(df.shape)
    print(df.memory_usage().sum())
    df.dropna(inplace=True)  # drop rows with not enough lookback (they will be nan)
```
The line `dropna` line is what throws the error

So when i run my code with adding 500 features/columns the output is following

```
(2177432, 503)
4398412768
```

However when i run the same exact code for 1000 features

```
(2177432, 1003)
8753276768
```

I get an error

```
  File "C:\Users\Ben\PycharmProjects\tradingbot\tradingbot\trainer\sample_maker.py", line 17, in get_signal_features
    df.dropna(inplace=True)  # drop rows with not enough lookback (they will be nan)
  File "C:\Users\Ben\PycharmProjects\tradingbot\venv\lib\site-packages\pandas\core\frame.py", line 4751, in dropna
    count = agg_obj.count(axis=agg_axis)
  File "C:\Users\Ben\PycharmProjects\tradingbot\venv\lib\site-packages\pandas\core\frame.py", line 7807, in count
    return result.astype("int64")
  File "C:\Users\Ben\PycharmProjects\tradingbot\venv\lib\site-packages\pandas\core\generic.py", line 5698, in astype
    new_data = self._data.astype(dtype=dtype, copy=copy, errors=errors)
  File "C:\Users\Ben\PycharmProjects\tradingbot\venv\lib\site-packages\pandas\core\internals\managers.py", line 582, in astype
    return self.apply("astype", dtype=dtype, copy=copy, errors=errors)
  File "C:\Users\Ben\PycharmProjects\tradingbot\venv\lib\site-packages\pandas\core\internals\managers.py", line 442, in apply
    applied = getattr(b, f)(**kwargs)
  File "C:\Users\Ben\PycharmProjects\tradingbot\venv\lib\site-packages\pandas\core\internals\blocks.py", line 625, in astype
    values = astype_nansafe(vals1d, dtype, copy=True)
  File "C:\Users\Ben\PycharmProjects\tradingbot\venv\lib\site-packages\pandas\core\dtypes\cast.py", line 868, in astype_nansafe
    raise ValueError("Cannot convert non-finite values (NA or inf) to integer")
ValueError: Cannot convert non-finite values (NA or inf) to integer
```

The version i am running is 1.0.5

### PR DESCRIPTION
**Pull request by ben-arnao**
- [ ] closes #35227
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
Can you add a test for the behavior you are trying to fix? As reviewers that's typically the first thing we look at

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -1313,7 +1313,7 @@ def check_below_min_count(
             non_nulls = np.prod(shape)
         else:
             non_nulls = mask.size - mask.sum()
-        if non_nulls < min_count:
+        if non_nulls > 0 and non_nulls < min_count:
```
Comment (simonjayhawkins) [inline pandas/core/nanops.py:1316]: see https://github.com/pandas-dev/pandas/issues/35227#issuecomment-663492635

**Comment by TomAugspurger** [comment]
@ben-arnao can you update here in response to comments?

**Comment by ben-arnao** [comment]
@TomAugspurger I wasn't sure if this was something people wasn't to continue with. I can make the changes @simonjayhawkins suggested if that's what's needed.

**Comment by arw2019** [comment]
Closing for now. @ben-arnao ping us whenever you'd like to continue and we'll reopen!

**Pull request closed without merging**